src/main/java/org/killbill/billing/plugin/adyen/api/ExpiredPaymentPolicy.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2014-2016 Groupon, Inc
+ * Copyright 2014-2016 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.plugin.adyen.api;
+
+import java.util.List;
+
+import org.joda.time.DateTime;
+import org.killbill.billing.payment.api.TransactionType;
+import org.killbill.billing.payment.plugin.api.PaymentPluginStatus;
+import org.killbill.billing.payment.plugin.api.PaymentTransactionInfoPlugin;
+import org.killbill.billing.plugin.adyen.client.AdyenConfigProperties;
+import org.killbill.clock.Clock;
+
+public class ExpiredPaymentPolicy {
+
+    private final Clock clock;
+
+    private final AdyenConfigProperties adyenProperties;
+
+    public ExpiredPaymentPolicy(final Clock clock, AdyenConfigProperties adyenProperties) {
+        this.clock = clock;
+        this.adyenProperties = adyenProperties;
+    }
+
+    public boolean isExpired(List<PaymentTransactionInfoPlugin> paymentTransactions) {
+        // Expecting only one payment transaction
```

Comment:
I think looking at initial transactions only is fine (stuck PENDING captures should probably not be cancelled).

However, it is possible to have things like AUTH failed then AUTH pending for the same payment. The state machine should ensure that there is only one pending transaction though, so looking at the last transaction below should be safe.